addons/mail/static/src/chatter/web_portal/chatter.js

```diff
@@ -274,11 +272,12 @@ export class Chatter extends Component {
      * @param {['activities'|'followers'|'attachments'|'messages'|'suggestedRecipients']} requestList
      */
     load(thread, requestList = ["followers", "attachments", "messages", "suggestedRecipients"]) {
-        if (!thread.id || !this.state.thread?.eq(thread)) {
+        if (!this.state.thread?.eq(thread)) {
             return;
         }
-        if (this.props.hasActivities && !requestList.includes("activities")) {
-            requestList.push("activities");
+        if (!thread.id) {
+            this.store.fetchData({ thread_has_activities: { model: thread.model } });
```

Comment:
Instead of an extra RPC, it should come from the data the web client is fetching about a model.

And it should not be stored on each thread, but on its model only.